### DISCUSSION TITLE
Possible to custom fingerprint method or introduce new customisable method

### DISCUSSION THREAD

**Discussion by sts-ryan-holton** [laravel/framework#48065] [Ideas]
Hi :wave:

I'm working on a Laravel 10 project and am storing the generated request `fingerprint` in my database. I'm using Laravel separately from my front-end (Nuxt front-end) and need to modify the fingerprint data used to construct it.

Here's the current implementation:

```php
/**
 * Get a unique fingerprint for the request / route / IP address.
 *
 * @return string
 *
 * @throws \RuntimeException
 */
public function fingerprint()
{
    if (! $route = $this->route()) {
        throw new RuntimeException('Unable to generate fingerprint. Route unavailable.');
    }

    return sha1(implode('|', array_merge(
        $route->methods(),
        [$route->getDomain(), $route->uri(), $this->ip()]
    )));
}
```

I was wondering whether, either:

- Introduce a optional array argument to add additional stuff to the request, e.g:

```php
// foo and bar could be any value
$request->fingerprint(['foo', 'bar'])
```

or...

- introduce a new `fingerprint` method such as:

```php
$request->customFingerprint()
```

Where I could customise the data.

**Comment by henzeb** [comment]
I'd say go with the extra parameter option in fingerprint. I doubt adding an extra, optional, parameter is a breaking change as there is no interface involved here. 

Just so you know: you don't have to copy over code in here...just point at the code with a permalink: https://github.com/laravel/framework/blob/c1e62815ada14aa1c4c8362572572a92bc3166b1/src/Illuminate/Http/Request.php#L627-L644

**Reply by sts-ryan-holton** [reply]
Sounds good to me 👍
